pyccel/ast/core.py

```diff
@@ -2271,6 +2210,67 @@ def prefix(self):
         """
         return self._prefix
 
+class ConstructorCall(DottedFunctionCall):
+
+    """
+    It  serves as a constructor for undefined function classes.
+
+    Parameters
+    ----------
+    func: FunctionDef, str
+        an instance of FunctionDef or function name
+
+    arguments: list, tuple, None
+        a list of arguments.
+
+    """
+    __slots__ = ('_cls_variable', '_func', '_arguments')
+    _attribute_nodes = ('_func', '_arguments')
+
+    is_commutative = True
+
+    # TODO improve
+
+    def __init__(
+        self,
+        func,
+        arguments,
+        cls_variable=None,
+        ):
+        if not isinstance(func, (FunctionDef, Interface, str)):
+            raise TypeError('Expecting func to be a FunctionDef or str')
+
+        self._cls_variable = cls_variable
+        self._func = func
+        self._arguments = (FunctionCallArgument(cls_variable),) + arguments
+        super().__init__(self._func, self._arguments, self._cls_variable)
+
+    def __str__(self):
+        name = str(self.name)
+        args = ''
+        if not self.arguments is None:
+            args = ', '.join(str(i) for i in self.arguments)
+        return '{0}({1})'.format(name, args)
+
+    @property
+    def func(self):
+        return self._func
```

Comment:
Can this be inherited from the superclass?

Reply:
fixed